### PR TITLE
Edit to fix increacing memory usage

### PR DESCRIPTION
**Pull request by hrmhatef**
### What was the problem?

- This PR resolves memory increment which is described in issue #88
- There is some memory leaks on `instruments` so the issue should be still open.

### How was it solved?

- Implement `close` functions for `ReadWriter` to close the connection.
- Define an empty function for  `StateWriter` to call it from JS.
- Finally implement `close` function for `StateReadWriter` in `state_db.js` to release the memory.

### How was it tested?

- Add a new unit test to check close function in JS.
- Also add a new unit test to check multi_thread usage for StateWriter.
- All tests passed.

### Additional info:

- The result of the `Instruments` for `state_long.js` shows memory usage will be drop when GC is running by JS.
- It depends on the running machine (also the number of iterations for `inner` and `outer` loops) but it should be remain fixed during the execution when the GC is called.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -40,6 +40,9 @@ version = "0.4.3"
 [dependencies.tempdir]
 version = "0.3"
 
+[dependencies.rand]
+version = "0.8.5"
```
Comment (shuse2) [inline Cargo.toml:44]: shoulnd't this dev dependency?

**Review by shuse2 — CHANGES_REQUESTED** [review]

**Review by matjazv — APPROVED** [review]
Great work @hrmhatef!
This was a tough one.

File: Cargo.toml (hunk 1 above)
Comment (hrmhatef) [reply Cargo.toml]: Done.

**Review by shuse2 — APPROVED** [review]

**Comment by matjazv** [comment]
In `yarn.lock` json5 is degraded from `json5@^2.2.3` to` json5@^2.2.1`.

**Comment by shuse2** [comment]
@matjazv I think now it doesn't include the yarn.lock change

**Pull request merged**